### ISSUE TITLE
[TODO]: v2 Improve initial featurization

### ISSUE THREAD

**Issue by oscarwumit** [chemprop/chemprop#547]
**Notes**
Implement better initial featurization algorithm. For example, atomic numbers and formal charges are one-hot-encoded in v1, but some benchmark suggests that change to ordinal/sequential encoding could accelerate training and achieve similar or better model performance, for one-hot-encoding tends to demand more memory usage and numerical values of formal charges do have associated chemical meaning.

Will think of a implementation that allows users to choose encoding methods for initial featurization (default to new settings but allows for v1 option).

Will also add some additional features that might benefit radicals, charged species, etc.

**Comment by davidegraff** [comment]
I've been wondering about this as well. Could we think about potentially splitting up the atom featurization into `label` + "additional features" where `label` is fed through an `nn.Embedding(NUM_ATOMS)` and then `torch.cat`-ed to the additional features?
